src/plugins/controls/public/options_list/embeddable/options_list_embeddable.tsx

```diff
@@ -307,27 +310,34 @@ export class OptionsListEmbeddable extends Embeddable<OptionsListEmbeddableInput
               mode: 'absolute' as 'absolute',
             }
           : globalTimeRange;
-      const { suggestions, invalidSelections, totalCardinality, rejected } =
-        await this.optionsListService.runOptionsListRequest(
-          {
-            sort,
-            field,
-            query,
-            filters,
-            dataView,
-            timeRange,
-            searchString: searchString.value,
-            runPastTimeout,
-            selectedOptions,
-          },
-          this.abortController.signal
-        );
-      if (rejected) {
-        // This prevents a rejected request (which can happen, for example, when a user types a search string too quickly)
-        // from prematurely setting loading to `false` and updating the suggestions to show "No results"
+
+      const response = await this.optionsListService.runOptionsListRequest(
+        {
+          sort,
+          size,
+          field,
+          query,
+          filters,
+          dataView,
+          timeRange,
+          runPastTimeout,
+          selectedOptions,
+          allowExpensiveQueries,
+          searchString: searchString.value,
+        },
+        this.abortController.signal
+      );
+      if (this.optionsListService.optionsListResponseWasFailure(response)) {
+        if (response.error === 'aborted') {
+          // This prevents an aborted request (which can happen, for example, when a user types a search string too quickly)
+          // from prematurely setting loading to `false` and updating the suggestions to show "No results"
+          return;
+        }
+        this.onFatalError(response.error);
```

Comment:
If the response was a failure but it wasn't aborted, that means that something else happened on the server side - for example, the network connection was lost, the request was somehow malformed, etc. So, throw a fatal error in this case.